linta-android/src/main/java/com/swvl/lint/checks/DuplicateResourceFilesDetector.kt

```diff
@@ -56,39 +56,42 @@ class DuplicateResourceFilesDetector : ResourceXmlDetector() {
             .transform(DOMSource(document), StreamResult(stringWriter))
 
         currentDocument = stringWriter.toString()
+        stringWriter.flush()
 
         removeToolsNamespaceAttributes(document.firstChild ?: return)
 
+        // Remove whitespaces.
         currentDocument = currentDocument.replace("\\s+".toRegex(), "")
 
+        // Cache the document.
         resources[currentDocument] =
             resources.getOrDefault(currentDocument, ArrayList()).apply {
                 add(ResourceDeclaration(context.file.name, context.createLocationHandle(document)))
             }
     }
 
-    private fun removeToolsNamespaceAttributes(child: Node?) {
-        if (child?.childNodes?.length == 0) {
-            return
+    private fun removeToolsNamespaceAttributes(node: Node?) {
+        // Remove tools namespace and all attributes under it.
+        val attributesCount = node?.attributes?.length ?: 0
+        for (i in 0 until attributesCount) {
+            val attr = node?.attributes?.item(i)
+            if (attr?.namespaceURI == TOOLS_NAMESPACE_URI || attr?.nodeValue == TOOLS_NAMESPACE_URI) {
+                currentDocument = currentDocument.replace(attr.toString(), "")
+            }
         }
 
-        val childrenCount = child?.childNodes?.length ?: 0
+        // Do the same with all children.
+        val childrenCount = node?.childNodes?.length ?: 0
         for (i in 0 until childrenCount) {
-            val subChild = child?.childNodes?.item(i)
-            val attributesCount = subChild?.attributes?.length ?: 0
-            for (j in 0 until attributesCount) {
-                val attr = subChild?.attributes?.item(j)
-                if (attr?.namespaceURI == TOOLS_NAMESPACE_URI)
-                    currentDocument = currentDocument.replace(
-                        attr.toString(),
-                        ""
-                    )
-            }
-            removeToolsNamespaceAttributes(subChild)
+            val child = node?.childNodes?.item(i)
+            removeToolsNamespaceAttributes(child)
         }
     }
 
     override fun afterCheckRootProject(context: Context) {
+        // Clear the last document.
+        currentDocument = ""
+
```

Comment:
why do we need to do this?

Reply:
It was just to release some memory, since it's not needed anymore, but anyway I refactored the code now and removed this part.